source/development/packaging/generate-indexer-package.rst

```diff
@@ -0,0 +1,314 @@
+
+Assembly stage
+^^^^^^^^^^^^^^
+
+Docker environment
+-------------------
+
+Pre-requisites:
+
+-  Current directory: ``wazuh-indexer/``
+-  Existing package in ``wazuh-indexer/artifacts/dist/{rpm|deb}``, as a
+   result of the *Build* stage.
+-  Using the `Docker
+   environment <https://www.github.com/wazuh/wazuh-indexer/tree/master/docker>`__:
+
+ .. tabs::
+
+   .. group-tab:: RPM
+
+      .. code-block:: console
+
+         # docker exec -it wi-assemble_|WAZUH_CURRENT| bash packaging_scripts/assemble.sh -a x64 -d rpm
+
+   .. group-tab:: DEB
+
+      .. code-block:: console
+
+         # docker exec -it wi-assemble_|WAZUH_CURRENT| bash packaging_scripts/assemble.sh -a x64 -d deb
+   
+   .. group-tab:: TAR
+
+      .. code-block:: console
+
+         # docker exec -it wi-assemble_|WAZUH_CURRENT| bash packaging_scripts/assemble.sh -a x64 -d tar
+
+
+Local package generation
+------------------------
+
+**Note:** set the environment variable ``TEST=true`` to assemble a
+package with a minimal set of plugins, speeding up the assembly process.
+
+.. tabs::
+
+   .. group-tab:: RPM
+
+      The ``assemble.sh`` script will use the output from the ``build.sh``
+      script and use it as a base to bundle together a final package
+      containing the plugins, the production configuration and the service
+      files.
+      
+      The script will:
+      
+      #. Extract the RPM package using ``rpm2cpio`` and ``cpio`` tools.
+      
+            By default, ``rpm2cpio`` and ``cpio`` tools expect the package to
+            be in ``wazuh-indexer/artifacts/tmp/rpm``.The script takes care of
+            creating the required folder structure, copying also the min
+            package and the SPEC file.
+      
+         Current folder loadout at this stage:
+      
+         ::
+      
+            /rpm/$ARCH
+                /etc
+                /usr
+                /var
+                wazuh-indexer-min-*.rpm
+                wazuh-indexer.rpm.spec
+      
+         ``usr``, ``etc`` and ``var`` folders contain ``wazuh-indexer`` files,
+         extracted from ``wazuh-indexer-min-*.rpm``.
+         ``wazuh-indexer.rpm.spec`` is copied over from
+         ``wazuh-indexer/distribution/packages/src/rpm/wazuh-indexer.rpm.spec``.
+         The ``wazuh-indexer-performance-analyzer.service`` file is also
+         copied from the same folder. It is a dependency of the SPEC file.
+      
+      #. Install the plugins using the ``opensearch-plugin`` CLI tool.
+      
+      #. Set up configuration files.
+      
+            Included in ``min-package``. Default files are overwritten.
+      
+      #. Bundle an RPM file with ``rpmbuild`` and the SPEC file
+         ``wazuh-indexer.rpm.spec``.
+      
+            ``rpmbuild`` is part of the ``rpm`` OS package.
+      
+         ..
+      
+            ``rpmbuild`` is invoked from ``wazuh-indexer/artifacts/tmp/rpm``.
+            It creates the {BUILD,RPMS,SOURCES,SRPMS,SPECS,TMP} folders and
```

Comment:
`{BUILD,RPMS,SOURCES,SRPMS,SPECS,TMP}` seems to need text formatting such as code font style.